### PR TITLE
fix formatting of avatars in emails

### PR DESCRIPTION
**Pull request by macqueen**
@getsentry/python

### CODE REVIEW

File: src/sentry/utils/avatar.py [hunk 1]
```diff
@@ -88,3 +91,17 @@ def get_letter_avatar(display_name, identifier, size=None, use_svg=True):
             '{initials}</span>').format(color=color, initials=initials,
                                         size_attrs=size_attrs, font_size=font_size,
                                         line_height=line_height)
+
+
+def get_email_avatar(display_name, identifier, size=None):
+    try:
+        validate_email(identifier)
```
Comment (macqueen) [inline src/sentry/utils/avatar.py]: not sure if it's worth doing this or not. just figured it was one way to not even bother trying to get a gravatar if we don't have an email address.

File: src/sentry/utils/avatar.py [hunk 2]
```diff
@@ -88,3 +91,17 @@ def get_letter_avatar(display_name, identifier, size=None, use_svg=True):
...
+    except ValidationError:
+        pass
+    else:
+        resp = requests.get(get_gravatar_url(identifier, default=404))
```
Comment (mattrobenolt) [inline src/sentry/utils/avatar.py]: This can raise an exception, so I think we should be completely graceful here.

Comment (mattrobenolt) [inline src/sentry/utils/avatar.py]: Also, don't use `requests` directly. We have a wrapper `sentry.http:safe_urlopen` that does some nicer things for us.

Comment (macqueen) [inline src/sentry/utils/avatar.py]: ok, just catch `requests.exceptions.RequestException`?

Comment (mattrobenolt) [inline src/sentry/utils/avatar.py]: I'd just catch `Exception`. There's no value in catching specific things, we just don't want anything to fail here ever.

Comment (macqueen) [inline src/sentry/utils/avatar.py]: ok

File: src/sentry/utils/avatar.py (hunk 1 above)
Comment (mattrobenolt) [inline src/sentry/utils/avatar.py]: I think this is fine.

**Comment by mattrobenolt** [comment]
I assume this renders fine. :+1:

**Comment by macqueen** [comment]
tested it using the PreviewBackend + forwarding it to myself and seems to (finally) be working.

**Pull request merged**